api/graphql/schema.graphql

```diff
@@ -39,8 +39,10 @@ enum LegacyRole {
 
 type User {
   id: ID!
-  sub: String
-  legacyRoles: [LegacyRole] @rename(attribute: "legacy_roles")
+  sub: String @canAccess(requiredRoles: ["platform_admin"], ownerKey: "id")
```

Comment:
I wish we could do this but as I understand it `@can` can only be used on top level items and not individual fields. That was why the `@canAccess` directive was introduced based off of [the documentation](https://lighthouse-php.com/5/security/authorization.html#custom-field-restrictions).

Or, are you suggesting to just update the new directive to pass a permission instead of a role?

Reply:
Ah, you're right, `@can` won't work. But we have the `@canOnParent` directive which we can use to apply specific policy methods to scalar fields.

Reply:
I'd love to hear @petertgiles's thoughts on the value of this.

Reply:
My confidence gets a bit shaky around GraphQL directives and the permission model.  :grimacing: Could the HasRolesDirective be reconfigured as HasPermissionsDirective to be similar to the `can` directive but still work at the field level?

Reply:
Yeah, we can pass anything into the directive to help us determine if the user has access. I think that makes sense to, we seem to always check permissions rather than roles as a whole. Though, that could inflate the number of permissions we have (not sure that is a bad thing, just more to maintain).

Reply:
Okay, that sounds like a good compromise. Let's replace the hasRoles directive with one that checks for a permission. (perhaps with a variation or arg for team-based permissions?) That way, our access controls are still documented in permissions, but we also have something a bit more direct than policy methods for straightforward permission checks.

Reply:
Alright, based on my testing, I believe I have this working. I'm not sure what permissions to use/introduce for now (I think we need to make new ones to be more granular). I'm wondering (assuming it is working) if what I implemented makes sense 🤔 

[user permissions instead of roles for directive](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7173/commits/b43543494d78f4f63ce0757e0a81dcf06fde482e)